widget/discrepancy-note/dn-widget.js

```diff
@@ -574,12 +574,28 @@ class Comment extends Widget {
             widget.style[ o[ 0 ] ] = o[ 1 ];
         } );
 
-        // Scroll directly to Query modal if not enough height for user to view the Query modal
-        // https://github.com/OpenClinica/enketo-express-oc/issues/378
-        if ( this.linkedQuestion.classList.contains( 'or-appearance-image-map' ) ) {
-            setTimeout( () => {
-                this._scrollToview();
-            }, 500 );
+        // https://github.com/OpenClinica/enketo-express-oc/issues/495
+        const maps = document.querySelectorAll( '.or-appearance-image-map' ),
```

Comment:
CMIIW. yes, because I think we need to reconsider every map is fully loaded before scrolling the DN modal to the correct position and if we use `this.linkedQuestion` we may not detect the SVG if we open DN on non-image map question. Also, you're correct about this one

> Is it possible for the SVG to be created before this tracker starts tracking (and _scrollToview never gets called)

sorry, I thought that for auto open DN but that issue for another DN after all SVG is loaded. So I plan to make the changes like this

````
   _mapChecker() {
        const root = this.element.closest( 'form' ),
        maps = root.querySelectorAll( '.or-appearance-image-map' );
        let ready = true;
        maps.forEach( map => {
            if ( !map.querySelector( 'svg' ) )
                ready = false;
        } );

        let count = 0;
        if ( !ready ) {
            const mutationObserver = new MutationObserver( mutations => {
                mutations.forEach( mutation => {
                    if ( mutation.attributeName === 'viewBox' && mutation.target.tagName === 'svg' ) {
                        count++;
                        if ( count === maps.length ) {
                            this._scrollToview();
                            mutationObserver.disconnect();
                        }
                    }
                });
            });

            mutationObserver.observe( root, {
                attributes: true,
                childList: true,
                subtree: true });
        } else {
            this._scrollToview();
        }
    }
````
is that acceptable or any thought about it? Thanks

Reply:
> CMIIW. yes, because I think we need to reconsider every map is fully loaded before scrolling the DN modal to the correct position

Ah ok. In that case, I think you can use a top-level variable to ensure the maps are only tracked once (even if the form has 100 image-map questions).

> code for ready variable

I like using a one-liner `let ready = maps.every( map => map.querySelector('svg'))` for this.

Reply:
Code updated but I'm not sure about 
> I think you can use a top-level variable to ensure the maps are only tracked once (even if the form has 100 image-map questions)

With the current code, the observer only runs once but the maps checking will run every time DN is opened.  I tried to set the `this.ready` but that only works on the current widget, I don't know how to set it just once, and can be used for all DN widgets.